src/main/java/org/apache/sysml/hops/AggUnaryOp.java

```diff
@@ -329,8 +329,15 @@ public boolean allowsAllExecTypes()
 
 	@Override
 	protected double computeOutputMemEstimate( long dim1, long dim2, long nnz )
-	{		
-		double sparsity = OptimizerUtils.getSparsity(dim1, dim2, nnz);
+	{
+		double sparsity = -1;
+		if (DMLScript.USE_ACCELERATOR) {
+			// The GPU version (for the time being) only does dense outputs
+			sparsity = 1.0;
+		} else {
+			sparsity = OptimizerUtils.getSparsity(dim1, dim2, nnz);
+		}
```

Comment:
I would recommend to use the same output memory estimate (in order to ensure that the compilation of subsequent operations is not affected) and account for the dense block only in the intermediate requirements for GPUs.